integration-tests/proxy_test.go

```diff
@@ -437,3 +437,106 @@ func TestProxyIngressAndEgress(t *testing.T) {
+		"--ingress", "udp:4242:127.0.0.1:4242",
+		"--ingress", "udp:4242:127.0.0.2:4243"})
+	require.Error(err)
+
+	// duplicate udp egress port
+	_, err = helper.containerExec(ctx, node1, []string{"nexd", "--username", username, "--password", password, "proxy",
+		"--egress", "udp:4242:100.100.0.1:4242",
+		"--egress", "udp:4242:100.100.0.2:4243"})
+	require.Error(err)
+
+	// Invalid ingress tcp listen port
+	_, err = helper.containerExec(ctx, node1, []string{"nexd", "--username", username, "--password", password, "proxy",
+		"--ingress", "tcp:90000:127.0.0.1:8080"})
+	require.Error(err)
+
+	// Invalid egress tcp listen port
+	_, err = helper.containerExec(ctx, node1, []string{"nexd", "--username", username, "--password", password, "proxy",
+		"--egress", "tcp:90000:100.100.0.1:8080"})
+	require.Error(err)
+
+	// Invalid ingress udp listen port
+	_, err = helper.containerExec(ctx, node1, []string{"nexd", "--username", username, "--password", password, "proxy",
+		"--ingress", "udp:90000:127.0.0.1:4242"})
+	require.Error(err)
+
+	// Invalid egress udp listen port
+	_, err = helper.containerExec(ctx, node1, []string{"nexd", "--username", username, "--password", password, "proxy",
+		"--egress", "udp:0:100.100.0.1:4242"})
+	require.Error(err)
+
+	// Invalid ingress tcp connect port
+	_, err = helper.containerExec(ctx, node1, []string{"nexd", "--username", username, "--password", password, "proxy",
+		"--ingress", "tcp:8080:127.0.0.1:90000"})
+	require.Error(err)
+
+	// Invalid egress tcp connect port
+	_, err = helper.containerExec(ctx, node1, []string{"nexd", "--username", username, "--password", password, "proxy",
+		"--egress", "tcp:8080:100.100.0.1:0"})
+	require.Error(err)
+
+	// Invalid ingress udp connect port
+	_, err = helper.containerExec(ctx, node1, []string{"nexd", "--username", username, "--password", password, "proxy",
+		"--ingress", "udp:4242:127.0.0.1:90000"})
+	require.Error(err)
+
+	// Invalid egress udp connect port
+	_, err = helper.containerExec(ctx, node1, []string{"nexd", "--username", username, "--password", password, "proxy",
+		"--egress", "udp:4242:100.100.0.1:0"})
+	require.Error(err)
+
+	// Invalid ingress protocol
+	_, err = helper.containerExec(ctx, node1, []string{"nexd", "--username", username, "--password", password, "proxy",
+		"--ingress", "tcpa:8080:127.0.0.1:80"})
+	require.Error(err)
+
+	// Invalid egress protocol
+	_, err = helper.containerExec(ctx, node1, []string{"nexd", "--username", username, "--password", password, "proxy",
+		"--egress", "tcpa:8080:100.100.0.1:80"})
+	require.Error(err)
+
+	// Incomplete ingress rule
+	_, err = helper.containerExec(ctx, node1, []string{"nexd", "--username", username, "--password", password, "proxy",
+		"--ingress", "tcp:8080"})
+	require.Error(err)
+
+	// Incomplete egress rule
+	_, err = helper.containerExec(ctx, node1, []string{"nexd", "--username", username, "--password", password, "proxy",
```

Comment:
I think taking out this string array (given it's exactly same parameter) and declare it at the start of the test would clean the containerExec across the test ?

Reply:
done, I put the common args in a new `baseArgs` slice at the beginning